crates/faucet/src/lib.rs

```diff
@@ -144,7 +152,7 @@ impl Faucet {
             Some(url) => Arc::new(RemoteTransactionProver::new(url)),
             None => Arc::new(LocalTransactionProver::default()),
         };
-        let id = FaucetId::new(account.id(), network_id);
+        let id = FaucetId::new(account.id(), endpoint.to_network_id());
```

Comment:
I think this is fine for now, but mapping endpoint to a network ID may be fragile (e.g., in theory just an IP address should work as an endpoint). So, we may need to specify the network ID regardless.

Let's create an issue for this.